tests/unit_tests/scheduler/test_slurm_driver.py

```diff
@@ -85,6 +85,26 @@ async def test_memory_is_set(memory_in_mb):
     )
 
 
+@pytest.mark.usefixtures("capturing_sbatch")
+@given(memory_in_bytes=st.integers(min_value=1))
+async def test_realization_memoryt(memory_in_bytes):
+    driver = SlurmDriver(realization_memory=memory_in_bytes)
+    await driver.submit(0, "sleep", name="myjobname")
+    assert f"--mem={memory_in_bytes // 1024**2}M" in Path(
+        "captured_sbatch_args"
+    ).read_text(encoding="utf-8")
+
+
+async def test_overspecified_memory_allocation():
+    # Emptry string for memory is equivalent to zero integer for realiation_memory
+    SlurmDriver(memory="", realization_memory=1)
+    SlurmDriver(memory="1mb", realization_memory=0)
+    with pytest.raises(ValueError):
+        SlurmDriver(memory="1mb", realization_memory=1)
+    with pytest.raises(ValueError):
+        SlurmDriver(memory="0", realization_memory=1)
```

Comment:
Should this be input as pytest parameters instead? Here we are testing 4 things at the same time

Reply:
Fixed!